sdk/appconfiguration/azure-spring-cloud-feature-management/README.md

```diff
@@ -259,6 +259,104 @@ Options are available to customize how targeting evaluation is performed across
     }
 ```
 
+## Dynamic Features
+
+Using `FeatureManager`'s `getVariantAsync` method allows for returning of dynamic features based off a feature filter. Usage:
+
+```java
+featureManager.getVariantAsync("DiscountBanner", DiscountBanner.class)
+```
+
+This example shows requesting a Dynamic Feature named "DiscountBanner", it requires the name of the Dynamic Feature and the type that needs to be returned.
+
+### Defining a Dynamic Feature
+
+A Dynamic Feature can be defined in the local configuration file using the following format.
+
+```yaml
+feature-management:
+  DiscountBanner:
+    assigner: "Microsoft.Targeting"
+    variants:
+      -
+        name: Big
+        default: true
+        configuration-reference: "DiscountBanner:Big"
+        assignment-parameters:
+          Audience:
+            users:
+              - Jeff
+              - Alicia
+            groups:
+              -
+                 name: "Ring0"
+                 rolloutPercentage: 80
+              -
+                 name: "Ring1"
+                 rolloutPercentage: 50
+            defaultRolloutPercentage: 20
+      -
+        name: Small
+        default: false
+        configuration-reference: "DiscountBanner:Small"
+        assignment-parameters:
+          Audience:
+            users:
+              - "Jack"
+              - "Amanda"
+            groups:
+              -
+                 name: "Ring0"
+                 rolloutPercentage: 10
+              -
+                 name: "Ring1"
+                 rolloutPercentage: 2W0
+            defaultRolloutPercentage: 0
+feature-variant:
+  DiscountBanner:
+    Big:
+      size: 400
+      color: "#BBB"
+    Small:
+      size: 150
+      color: "#DDD"
+```
+
+A Dynamic Feature has two main parts the assigner and a list of variants. The assigner in this case `feature-management.DiscountBanner.assigner` is pointing the `FeatureFilter` `Microsoft.Targeting` that will evaluate the list of variants and choose which configuration-reference will be used. Each variant has; a name, configuration-reference, and assignment-parameters, and one of them needs to be set as the default variant. The name field is just meta data to reference the variant and just needs to be unique. The configuration-reference is a link to the configurations defined under `feature-variant` in the configuration file. **Note:** the colon notation is required when denoting parts of the configuration. If it is needed multiple colons can be used to reference sub configurations. i.e. `DiscountBanner:Big:size` would return the `Integer` 400.
```

Comment:
Should have comma in first sentence.

`Microsoft.Targeting` in this case is an assigner not a feature filter.

Why a semi-colon in "has;" ?

I would change the ", and one of them needs to be set as the default variant." to it's own sentence. "One of the listed variants needs to be configured as the default variant for the feature."

The "just" usage in the "The name field is just..." sentence should be removed.

Calling the configuration reference a "link" might be confusing. Link usually refers to URL.

Reply:
It may be worth it to have a bulleted list of the properties to define their usage/requirements.